libraries/lib-realtime-effects/RealtimeEffectList.cpp

```diff
@@ -25,7 +25,7 @@ std::unique_ptr<ClientData::Cloneable<>> RealtimeEffectList::Clone() const
 {
    auto result = std::make_unique<RealtimeEffectList>();
    for (auto &pState : mStates)
-      result->mStates.push_back(pState);
+      result->mStates.push_back(pState->Clone());
```

Comment:
I wonder if it breaks the undo/redo stack. I remember that shallow copy was used there for a reason. @dozzzzer can you please check?

Reply:
A good point.

Define what "break" means exactly, but there may be some other visible change of behavior.  This code will also be visited when pushing undo items.

Undo items in memory may occupy more storage.

The state of sliders in dialogs, although not causing an undo push or pop when you change them directly, will be snap-shotted whenever the undo stack is pushed for other reasons, and restored to previous state when you undo.

I had raised the question whether this would be desired and intended undo/redo behavior or not.  It was decided, not.  But now, this change would make it so.

So there may be more work to pass more information about how `Clone()` is reached, and change its behavior only in the case you get there from the Duplicate editing command.

Reply:
Consider this PR just draft, then.  The good new is that the cause of the reported bug is known and easily fixable but for the other side effects of the fix.

Reply:
As far as I remember the problem is that UI state is "pinned" to a specific `RealtimeEffectState`. So at some point UI modifies the state which is stored in the history, not the one used by the processor. 

This might be no longer a problem, as we've made a lot of changes on top. We need to regress all effect families carefully, specifically when using Undo/Redo with UI open.